### ISSUE TITLE
"Placed" Candidates not getting filtered out of requests

### ISSUE THREAD

**Issue by Jerryescandon** [GCTC-NTGC/gc-digital-talent#4834]
## Description

We have heard from people handling requests on the live site that the wrong people are showing up on requests (Status: Placed, Expired, etc)

I have tried reproducing on UAT but was unable to. But seems to be happening on prod.

```
I do the work around and get 7 people referred for a job, I look through the names and 4 people were already placed therefore I am only sending 3 CVs.
I went back and made a request to see what the manger saw on their side, and it said they were going to get 7 people but only 3 people are still available in the pool.
How come the placed candidates are still pulling?
```

## Acceptance Criteria
- [x] Step one - confirm with Anne-Marie's team whether/where this is still happening in Prod.
- [ ] Step two - if it is still happening, see if it can be reproduced on local.
- [ ] Step three - if its still a problem on local, either fix it or create a new ticket with a more detailed explanation.

   [image]: https://user-images.githubusercontent.com/305339/166570935-6425a615-82c7-4b2f-9840-5c7d698fa2ab.png
   [image-link]: https://example.com
   [figma]: https://go.talent.c4nada.ca/figma
   [sitemap]: https://go.talent.c4nada.ca/sitemap

**Comment by tristan-orourke** [comment]
@Jerryescandon can you double check if this is still an issue after v2.0.2 being deployed?
If it is still an issue... when they say they are seeing placed candidates, do they mean placed casual or placed permanent? Because we've been told to include PLACED_CASUAL candidates in the search results.

**Comment by Jerryescandon** [comment]
Yup we're aware, weve made sure to check for this in the past. We see not only placed permanent but all other too. Including expired. I will check with Mel in the morning

**Comment by tristan-orourke** [comment]
I've confirmed on Prod (which was updated recently), this is no longer an issue - or at least, it is not a problem on the public Search page, and #4672 will resolve this for admins.

My test:
I looked at a request in Prod. When I removed the Profile Complete filter (allowing admin_entry users to be matched) there were 20 users who matched the request - two of which did not have the QUALIFIED_AVAILABLE pool status.
When I recreated the filters on the public search page, it reported that 18 users matched my request - which matches my expected 20-2 number.
